aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -9,79 +9,108 @@
+    List,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+    cast,
+)
 
 from aiida import get_version, orm
 from aiida.common import json
 from aiida.common.exceptions import LicensingException
-from aiida.common.folders import RepositoryFolder, SandboxFolder, Folder
+from aiida.common.folders import Folder, RepositoryFolder, SandboxFolder
 from aiida.common.lang import type_check
-from aiida.common.log import override_log_formatter, LOG_LEVEL_REPORT
+from aiida.common.log import LOG_LEVEL_REPORT, override_log_formatter
 from aiida.orm.utils._repository import Repository
-
-from aiida.tools.importexport.common import exceptions, get_progress_bar, close_progress_bar
-from aiida.tools.importexport.common.config import EXPORT_VERSION, NODES_EXPORT_SUBFOLDER
-from aiida.tools.importexport.common.config import (
-    NODE_ENTITY_NAME, GROUP_ENTITY_NAME, COMPUTER_ENTITY_NAME, LOG_ENTITY_NAME, COMMENT_ENTITY_NAME
+from aiida.tools.importexport.common import (
+    close_progress_bar,
+    exceptions,
+    get_progress_bar,
 )
 from aiida.tools.importexport.common.config import (
-    get_all_fields_info, file_fields_to_model_fields, entity_names_to_entities, model_fields_to_file_fields
+    COMMENT_ENTITY_NAME,
+    COMPUTER_ENTITY_NAME,
+    EXPORT_VERSION,
+    GROUP_ENTITY_NAME,
+    LOG_ENTITY_NAME,
+    NODE_ENTITY_NAME,
+    NODES_EXPORT_SUBFOLDER,
+    entity_names_to_entities,
+    file_fields_to_model_fields,
+    get_all_fields_info,
+    model_fields_to_file_fields,
 )
 from aiida.tools.importexport.common.utils import export_shard_uuid
 from aiida.tools.importexport.dbexport.utils import (
-    check_licenses, fill_in_query, serialize_dict, check_process_nodes_sealed, summary, EXPORT_LOGGER, ExportFileFormat,
-    deprecated_parameters
+    EXPORT_LOGGER,
+    ExportFileFormat,
+    check_licenses,
+    check_process_nodes_sealed,
+    deprecated_parameters,
+    fill_in_query,
+    serialize_dict,
+    summary,
 )
-
-from .zip import ZipFolder
+from aiida.tools.importexport.dbexport.zip import ZipFolder
 
 __all__ = ('export', 'EXPORT_LOGGER', 'ExportFileFormat')
 
 
 def export(
-    entities=None,
-    filename=None,
-    file_format=ExportFileFormat.ZIP,
-    overwrite=False,
-    silent=False,
-    use_compression=True,
-    **kwargs
-):
+    entities: Optional[Iterable[Any]] = None,
+    filename: Optional[str] = None,
+    file_format: str = ExportFileFormat.ZIP,
+    overwrite: bool = False,
+    silent: bool = False,
+    use_compression: bool = True,
+    **traversal_rules: bool,
+) -> None:
     """Export AiiDA data
 
     .. deprecated:: 1.2.1
         Support for the parameters `what` and `outfile` will be removed in `v2.0.0`.
         Please use `entities` and `filename` instead, respectively.
 
-    :param entities: a list of entity instances; they can belong to different models/entities.
-    :type entities: list
+    :param entities: a list of entity instances;
+        they can belong to different models/entities.
```

Comment:
Am I wrong or does mypy somehow apply a shorter line length than the one we have configured?

Reply:
I don't think mypy dictates a line length, I think I just changed this when I was writing the code in https://github.com/aiidateam/aiida-archive, which dictates a line length of 88 (via black) which FYI does a lot better job of the formatting, especially for things like multi-line dictionaries

Reply:
Ok, but please don't do this in your PRs on aiida-core, it is not reviewer friendly.

If you want to propose this as a repo-wide policy, that is another thing, but this needs to be discussed.

Reply:
oh yeh well I wouldn't normally be writing the code in a different repo, its just that I only decided this should be upstreamed after I had written it all